@extends('layout.main')

@section('content')

    <div class="container-fluid py-4">
        <div class="d-flex justify-content-center">

            <div class="col-xl-9 col-lg-10 col-sm-12 mb-xl-0 mb-4">
                <div class="card">
                    <div class="card-header p-3 pt-2">
                        <div class="icon icon-lg icon-shape bg-gradient-primary shadow-primary text-center border-radius-xl mt-n4 position-absolute">
                            <i class="material-icons opacity-10">psychology</i>
                        </div>
                        <div class="text-end pt-1">
                            <h2 class="mb-0 text-info">spatial<span class="moffitt-text-blue">GE</span></h2>
                        </div>
                    </div>
                    <hr class="dark horizontal my-0">
                    <div class="card-body px-6 text-justify text-dark">

            <p><b>How to Get Started</b></p>

            <p >spatialGE provides a set of methods for the visualization
                and spatial analysis of gene expression from spatially-resolved transcriptomic
                experiments, such as those generated by the Visium, SMI, or MERFISH platforms.
                Users are not limited to those platforms: if gene counts and the spatial
                coordinates of each of those counts are available, they can be analyzed with
                spatialGE.
            </p>

            <p >To use spatialGE, users provide data in one of the following
                formats:</p>


            <p>1.From
                comma/tab separated files:</p>

            <p>a. Raw (untransformed) gene counts contained in comma- or tab-delimited files. One for
                each sample/tissue slice. The first column of each file contains gene names.
                Subsequent columns contain data for each spot or cell in the sample. If duplicate
                gene names are present, they will be added an appendix number (1, 2, etc).</p>

            <p><img width=362 height=209 id="Picture 1" src="how_to_get_started%202_files/image001.png"></p>

            <p>b.Coordinates (x, y) for each spot or cell in the sample contained in comma- or tab-delimited
                files. One per sample/tissue slice. The files must contain three columns:
                cell/spot IDs, y coordinates, and x coordinates. The cell/spot IDs must match
                the column names for each cell/spot (columns) in the RNA count files.
            </p>

            <p><img width=205 height=134 id="Picture 2" src="how_to_get_started%202_files/image002.png"></p>

            <p> 2. From space ranger outputs generated for the Visium platform:</p>

            <p>a.Gene counts in the form of .h5 files, one per sample/tissue slice
                (filtered_feature_bc_matrix.h5 in the figure below).</p>

            <p>b. Coordinate data as provided by space ranger (tissue_positions_list.csv in the figure below).</p>

            <p><img width=283 height=353 id="Picture 3" src="how_to_get_started%202_files/image003.jpg" ></p>


            <p>If the user has sample-level associated metadata (e.g.,
                patient of origin, therapy, overall survival), spatialGE will take this
                information and match it with each sample/tissue slice. This information is
                required for multi-sample comparison analytical modules.</p>

            <p ><b>FAQ</b></p>

            <p  style='text-indent:-.25in'><span
                    style='font-family:Symbol'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            </span></span><b>I have data from a GeoMx experiment. Can I analyze it with
                    spatialGE?</b></p>

            <p >Yes, you can. spatialGE can be applied to
                any spatially-resolved transcriptomics experiment as long as gene counts are
                associated to x and y coordinates within a tissue slice. Nevertheless, the
                methods in spatialGE have been designed to analyze spatial data that is densely
                sampled, and most GeoMx experiments are characterized by ROI-based sampling
                that often is sparse and targeting tissue compartments. As an example, while
                creating a gene expression surface (interpolation) based on 20 ROIs across a
                large tissue section is possible, it is likely not informative.</p>

            <p class=MsoListParagraphCxSpLast style='text-indent:-.25in'><span
                    style='font-family:Symbol'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            </span></span><b>Do I have to pay to use spatialGE?</b></p>

            <p  style='margin-left:.5in'>No, spatialGE is open to the
                research community. Users are only required to create an account, and to
                acknowledge its use by citing the research articles that describes the methods
                in spatialGE (see How to Cite section).</p>










                    </div>
                    <div class="card-footer p-3">
                        <p class="mb-0"><span class="text-success text-sm font-weight-bolder">footer</span></p>
                    </div>
                </div>
            </div>

        </div>




        <footer class="footer py-4  ">
            <div class="container-fluid">
                <div class="row align-items-center justify-content-lg-between">
                    <div class="col-lg-6 mb-lg-0 mb-4">
                        <div class="copyright text-center text-sm text-muted text-lg-start">
                            © 2023
                        </div>
                    </div>
                </div>
            </div>
        </footer>
    </div>


@endsection
